### ISSUE TITLE
“Editor is not responding” dialog should not be blocking

### ISSUE THREAD

**Issue by PointedEars** [atom/atom#16378]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
→ **This issue cannot be reproduced _reliably_ in Safe Mode, presumably because less packages are loaded then or some offending packages are not loaded. However, as in normal mode there are always an increasing number of packages to be loaded, some of which might have loading issues, as users grow to use Atom for an increasing number of purposes, this appears to be a general problem and should be solved in a general way.**
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
→ **general usability issue**
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
→ **After I reproduced the issue several times in normal, and checked in Safe Mode, I ran `atom --clear-window-state` without significant improvement.**
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
→ **search for `responding` AFAICS yields similar issues, but no suitable matches for the *general* problem with the current approach (CMIIW)**
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages
→ **not a case of missing functionality**

### Description

The “Editor is not responding” dialog is blocking Atom when it appears. The user is _forced_ to make a choice between “Force Close” and “Keep Waiting” to continue, even if the problem was only temporary.

### Steps to Reproduce

1. Start Atom, optionally when automatically restoring a session that has several windows with several tabs open (the test case had 5 windows, each with 5 to 10 editor/PDF viewer tabs open – maybe I am the Atom “power user” that nobody imagined :)).
2. Optionally, do something in Atom that takes a long time to complete.
3. Optionally, exit such an Atom session.

**Expected behavior:** When the editor is responding again, close the dialog automatically and continue as if nothing happened, as I have observed in the Chromium (57.0.2987.98) Web browser. [For QA, a non-modal message could still be displayed later, perhaps indicating which package(s) caused the delay and/or offer to submit a bug report.]

**Actual behavior:** Atom does not continue in _any_ window of a session until the user makes the choice for the _affected_ window. Especially, if the first window of a session to be restored is blocked by the dialog, all other windows remain _blank_ until the choice for the first window is made. Then the choice for the second window might need to be made, and so on.

In effect, if the user does not want to compromise their session by “Force Close”, they are forced, sometimes for each window and sometimes even repeatedly for one window, to select “Keep Waiting” in up to all windows of a session to get Atom working (again) or to terminate it cleanly.

**Reproduces how often:** _Always_, especially with many installed or with few or many slow loading/working packages, or many windows or many restored editor tabs per window.

### Versions

```
$ atom --version
Atom    : 1.23.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

$ apm --version
apm  1.18.11
npm  3.10.10
node 6.9.5 x64
atom 1.23.1
python 2.7.12+
git 2.11.0
```

**OS:** Debian/Devuan GNU+Linux jessie/stretch/sid

**But: This has probably been an issue since the first versions of Atom (which I used). With an increasing number of (useful) packages installed, and parallel work (in several windows) to be done, it just has now reached a point where the situation has become unbearable. Hence this bug report.**

## Installed Packages

```
$ apm list --installed
Community Packages (172) /home/[…]/.atom/packages
├── MagicPython@1.0.12
├── Zen@0.18.0
├── atom-beautify@0.30.9
├── atom-css-unit-converter@1.1.1
├── atom-mysql-snippets@0.3.1
├── atom-php-navigation@0.6.0
├── atom-ternjs@0.18.3
├── atom-x@0.4.0
├── atomatigit@1.5.5
├── autoclose-html@0.23.0
├── autocomplete-awk@0.1.2
├── autocomplete-bibtex@1.1.0
├── autocomplete-clang@0.11.4
├── autocomplete-html-entities@0.2.0
├── autocomplete-paths@2.12.1
├── autocomplete-php@0.3.7
├── autocomplete-python@1.10.5
├── autocomplete-xml@0.9.4
├── autoupdate-packages@1.3.1
├── block-comment@0.4.3
├── busy-signal@1.4.3
├── character-table@0.4.3
├── clipboard-plus@0.5.1
├── color-picker@2.2.5
├── column-select@1.0.1
├── compare-files@0.8.1
├── css-snippets@1.1.0
├── ctrl-last-tab@0.3.0
├── data-atom@0.26.2
├── docblockr@0.13.4
├── emmet@2.4.3
├── emmet-snippets-compatibility@1.0.6
├── es6-javascript@1.0.0
├── ex-mode@0.18.0
├── filesize@3.0.0
├── fixmyjs@1.4.1
├── fold-comments@0.6.0
├── git-blame@1.4.0
├── git-clone@0.2.0
├── git-control@0.9.0
├── git-diff-details@1.4.0
├── git-difftool@0.2.7
├── git-go@1.0.5
├── git-hide@1.0.0
├── git-history@3.3.0
├── git-log@0.4.1
├── git-plus@7.10.0
├── git-rebase@0.1.1
├── git-status@0.3.4
├── git-time-machine@1.5.9
├── gitignore-snippets@0.2.5
├── goto@1.8.3
├── goto-definition@1.3.4
├── haskell-grammar@0.4.0
├── highlight-selected@0.13.1
├── hyperclick@0.1.5
├── hyperclick-php@0.2.1
├── ide-java@0.7.0
├── ide-php@0.7.4 (disabled)
├── ide-typescript@0.7.2
├── indent-toggle-on-paste@0.1.3
├── intentions@1.1.5
├── iso8601-timestamp@0.1.1
├── javascript-snippets@1.2.1
├── js-hyperclick@1.12.2
├── js-refactor@0.7.5
├── language-apache@1.8.0
├── language-awk@0.1.12
├── language-babel@2.81.0
├── language-diff@0.7.0
├── language-gettext@0.6.1
├── language-ini@1.19.0
├── language-latex@1.1.1
├── language-lua@0.9.11
├── language-sql-mysql@0.3.1
├── language-swift@0.5.0
├── language-tcltk@0.1.5
├── latex-autocomplete@1.1.1
├── latex-completions@0.3.3
├── latex-friend@0.0.11
├── latex-hyperclick@0.2.0
├── latexer@0.3.0
├── latextools@0.8.5
├── lcars-syntax@0.0.1
├── less-autocompile@1.0.0
├── less-than-slash@0.17.0
├── line-count-status@0.0.3
├── line-jumper@0.14.1
├── linter@2.2.0 (disabled)
├── linter-checkbashisms@0.1.5
├── linter-chktex@1.3.1
├── linter-csslint@2.0.0
├── linter-htmlhint@1.4.0
├── linter-jshint@3.1.6
├── linter-less@2.6.0
├── linter-mixed-indent@1.1.1
├── linter-mypy@2.4.3
├── linter-php@1.5.1
├── linter-pylint@2.1.0
├── linter-shellcheck@1.4.6
├── linter-ui-default@1.6.10
├── linter-xmllint@1.4.3
├── local-history@4.3.1
├── man@0.8.2
├── merge-conflicts@1.4.5
├── method-separator@0.2.0
├── minimap@4.29.7
├── minimap-cursorline@0.2.0
├── minimap-find-and-replace@4.5.2
├── minimap-git-diff@4.3.1
├── minimap-hide@0.3.0
├── minimap-highlight-selected@4.6.1
├── minimap-linter@2.1.3
├── minimap-pigments@0.2.2
├── minimap-split-diff@0.3.7
├── multi-cursor@2.1.5
├── multi-cursor-plus@1.2.0
├── multi-file-open@0.1.1
├── multirow-tabs@0.3.3
├── navigate@0.1.17
├── navigate-indent@0.1.0
├── next-occurrence@0.5.0
├── nuclide@0.270.0
├── nuclide-format-js@0.0.43
├── open-git-modified-files@0.2.4
├── open-recent@5.0.0
├── open-this@0.5.0
├── pdf-view@0.61.0
├── php-debug@0.2.6
├── php-extract-method@0.2.1
├── php-getters-setters@0.5.5
├── pigments@0.40.2
├── platformio-ide-terminal@2.7.0
├── preview@0.17.0
├── preview-inline@1.5.1
├── project-folder@1.5.0
├── project-view@0.6.2
├── promote-folder-to-project@2.0.2
├── punchclock@0.6.5
├── python-isort@0.0.7
├── python-tools@0.6.9
├── qolor@0.4.2
├── quantum-shell@0.7.1
├── quick-query@0.12.2
├── recent-projects@0.13.1
├── refactor@0.11.5
├── regex-tester@0.2.0
├── remember-file-positions@0.2.3
├── remote-edit@1.9.0
├── revert-buffer@0.6.0
├── script@3.18.0
├── set-syntax@0.4.0
├── seti-ui@1.9.0
├── show-origin@0.2.0
├── sloc@1.1.0
├── sort-lines@0.18.0
├── split-diff@1.5.1
├── switch-header-source@0.24.4
├── symbols-tree-nav@0.15.5
├── symbols-tree-view@0.14.0
├── term@0.2.4
├── text-manipulation@0.6.0
├── todo-show@2.1.0
├── tool-bar@1.1.0
├── tree-view-git-status@1.4.0
├── up2date@0.1.2
├── vim-mode-clipboard-plus@0.2.0
├── vim-mode-plus@1.24.1
├── w3c-validation@0.4.0
├── wordcount@2.10.4
├── xkcd-comics@0.3.0
└── xml-tools@0.2.1
```

None of those packages are indicated to be incompatible.

**Comment by rsese** [comment]
Thanks for taking the time to contribute!  I think this is basically a duplicate of https://github.com/atom/atom/issues/11551 so you can subscribe there for updates if you'd like.

If I mis-understood anything, please let me know.

---

Because we treat our issues list as [the Atom team's backlog](https://en.wikipedia.org/wiki/Scrum_(software_development)#Product_backlog), we close duplicates to focus our work and not have to touch the same chunk of code for the same reason multiple times. This is also why we may mark something as duplicate that isn't an exact duplicate but is closely related.

For information on how to use GitHub's search feature to find out if something is a duplicate before filing, see [the How Can I Contribute? section of the Atom CONTRIBUTING guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-can-i-contribute).
